src/model/OWL.pl

```diff
@@ -382,13 +406,13 @@
   is_all_disjoint_classes(DC),
   tripledb_ask(DC,owl:members,RDF_list),
   is_rdf_list(RDF_list,List),
-  once((
+  (
     member(Sup_A,List), subclass_of(A,Sup_A),
     member(Sup_B,List), unify_disjoint_(B,Sup_B)
-  )).
+  ).
 
 unify_disjoint_(B,Disjoint) :-
-  ( var(B) -> B=Disjoint ; subclass_of(B,Disjoint) ).
+  ( B=Disjoint ; transitive(subclass_of(B,Disjoint)) ).
```

Comment:
I am not sure about this.

let's say var(B) holds, the second case would start generating all the subclasses of Disjoint (which is potentially a lot of classes in case B is a more general class such as Object).
It would be also redundant in a sense, because we already generated an answer that it holds "for each object", so we do not need to repeat it for each object subclass.

Reply:
any reason you want to have the transitive closure here?

Reply:
have you tried `transitive(has_disjoint_class(A,B))` ?

Reply:
> I am not sure about this.
> 
> let's say var(B) holds, the second case would start generating all the subclasses of Disjoint (which is potentially a lot of classes in case B is a more general class such as Object).
> It would be also redundant in a sense, because we already generated an answer that it holds "for each object", so we do not need to repeat it for each object subclass.

If B is unbound then the `subclass_of(B,Disjoint)` call is never called. Therefore e.g. this [testcase](https://github.com/sasjonge/knowrob/blob/8374cadd0db3bd6c2c4e776f22583666c644a6bd/src/model/OWL.plt#L464) fails, even though DisjCls1 and DisjCls2 (and therefore their subclasses) are disjoint.

> any reason you want to have the transitive closure here?

Disjointness transfers to all subclasses of a class, and not only direct subclasses.
 
> have you tried `transitive(has_disjoint_class(A,B))` ?

I'm not sure if this makes sense, because disjointness is not transitive.

Reply:
"Disjointness transfers to all subclasses of a class, and not only direct subclasses."

-- but that is a matter of reasoning, we are in the model section where reasoning should not take place with the exception of RDFS